purchases/src/main/kotlin/com/revenuecat/purchases/teststore/TestStoreBillingWrapper.kt

```diff
@@ -0,0 +1,257 @@
+            )
+        val storeProduct = testStorePurchasingData.storeProduct
+        showPurchaseDialog(activity, storeProduct, presentedOfferingContext)
+    }
+
+    override fun isConnected(): Boolean = connected
+
+    override fun queryPurchases(
+        appUserID: String,
+        onSuccess: (Map<String, StoreTransaction>) -> Unit,
+        onError: (PurchasesError) -> Unit,
+    ) {
+        debugLog { "TestStoreBillingAbstract: queryPurchases - returning cached purchases" }
+        onSuccess(purchaseCache)
+    }
+
+    override fun showInAppMessagesIfNeeded(
+        activity: Activity,
+        inAppMessageTypes: List<InAppMessageType>,
+        subscriptionStatusChange: () -> Unit,
+    ) {
+        debugLog { "TestStoreBillingAbstract: showInAppMessagesIfNeeded - no-op for test store" }
+    }
+
+    override fun getStorefront(
+        onSuccess: (String) -> Unit,
+        onError: PurchasesErrorCallback,
+    ) {
+        debugLog { "TestStoreBillingAbstract: getStorefront - returning US by default" }
+        onSuccess("US")
+    }
+
+    private fun showPurchaseDialog(
+        activity: Activity,
+        product: StoreProduct,
+        presentedOfferingContext: PresentedOfferingContext?,
+    ) {
+        val message = buildString {
+            append("Product: ${product.id}\n")
+            append("Price: ${product.price.formatted}\n")
+            product.defaultOption?.let { option ->
+                option.pricingPhases.forEach { phase ->
+                    append("Phase: ${phase.price.formatted} for ${phase.billingPeriod.iso8601}\n")
+                }
+            }
+        }
+
+        dialogHelper.showDialog(
+            activity = activity,
+            title = "Test Store Purchase",
+            message = message,
+            positiveButtonText = "Test valid Purchase",
+            negativeButtonText = "Test failed Purchase",
+            neutralButtonText = "Cancel",
+            onPositiveButtonClicked = {
+                completePurchase(product, presentedOfferingContext)
+            },
+            onNegativeButtonClicked = {
+                purchasesUpdatedListener?.onPurchasesFailedToUpdate(
+                    PurchasesError(
+                        PurchasesErrorCode.ProductNotAvailableForPurchaseError,
+                        "Simulated test purchase failure: no real transaction occurred",
+                    ),
+                )
+            },
+            onNeutralButtonClicked = {
+                purchasesUpdatedListener?.onPurchasesFailedToUpdate(
+                    PurchasesError(
+                        PurchasesErrorCode.PurchaseCancelledError,
+                        "Purchase cancelled by user",
+                    ),
+                )
+            },
+        )
+    }
+
+    private fun completePurchase(
+        product: StoreProduct,
+        presentedOfferingContext: PresentedOfferingContext?,
+    ) {
+        val purchaseToken = UUID.randomUUID().toString()
```

Comment:
Maybe we want to add some specific details in the token like user id or purchase date... For now just generating a random UUID.